### ISSUE TITLE
Audit the use of WAI-ARIA roles across USWDS

### ISSUE THREAD

**Issue by chandracarney** [uswds/uswds#6803]
### Is your request related to a problem? Please describe.
Some components might to be using old, incorrect, or may altogether lack [WAI-ARIA roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles).

We should audit the use of ARIA roles across USWDS with an eye towards semantic roles to investigate whether component roles need to be updated or need to be refactored. It was previously planned that a component might be refactored when building the Web Component version, but our initially planned (2024 to early 2025) Web Component timeline has been delayed. We might consider a refactor for components that pose issues for maintaining, or for known issues that would be directly addressed by adding semantic properties or refactoring.

While looking at ARIA roles throughout the system initially might be quick, we'll likely need a more detailed investigation on individual findings.

### Resources and known issues
- [W3C using application menus](https://www.w3.org/WAI/tutorials/menus/application-menus/)
- [If you use the WAI-ARIA role "application", please do so wisely!](https://www.marcozehe.de/if-you-use-the-wai-aria-role-application-please-do-so-wisely/)
- [Comment on previous finding in #5754 that alert should be refactored to use dialog](https://github.com/uswds/uswds/issues/5754#issuecomment-2666176641))

### Outcome
- [ ] Look at our use of ARIA roles across the USWDS and create a list of component roles
- [ ] Log the aria role in a new column in our [internal testing spreadsheet 🔒](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?gid=97481116#gid=97481116)
- [ ] Open issues for known/immediately identified refactor needs (known: date picker, alert)
- [ ] Open issues for further investigation on a component-level
